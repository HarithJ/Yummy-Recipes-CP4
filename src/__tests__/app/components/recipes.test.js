import React from 'react';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson, { shallowToJson } from 'enzyme-to-json';

import Recipes from '../../../components/recipes/recipes.js'

describe("recipes component", () => {
  global.localStorage.setItem("accessToken", "'sakjdhwdhu'")

  let pagination = {limit: 0, page: 1, totalPages: 1}
  const recipes = shallow(<Recipes pagination={pagination} categoryId={9}/>)

  it("renders properly", () => {
    expect(shallowToJson(recipes)).toMatchSnapshot();
  })

  it("renders all the recipes properly", () => {
    recipes.setState({recipes:[1,2,3,4,5]});

    expect(recipes.find("Recipe").length).toBe(5)
  })

  it("has a function to get recipes", () => {
    expect(recipes.instance().getRecipes());
  })

  it("has a function to get searched recipes", () => {
    pagination = {limit: 3}

    expect(recipes.instance().getSearchedRecipes("nameToBeSearched", pagination));
  })

  it("has a function to get searched recipes without pagination", () => {
    pagination = {limit: 0}

    expect(recipes.instance().getSearchedRecipes("nameToBeSearched", pagination));
  })

  it("has a function to get all recipes", () => {
    pagination = {limit: 3}
    expect(recipes.instance().getAllRecipes(pagination));
  })

  it("has a function to get all recipes without pagination", () => {
    pagination = {limit: 0}
    expect(recipes.instance().getAllRecipes(pagination));
  })

  it("has a function to delete a recipe", () => {
    expect(recipes.instance().handleDelete());
  })
})
