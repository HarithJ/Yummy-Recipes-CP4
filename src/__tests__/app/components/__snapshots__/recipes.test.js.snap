// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recipes component renders properly 1`] = `
<div>
  <AddRecipe
    categoryId={9}
    getRecipes={[Function]}
  />
</div>
`;
